tracer/src/Datadog.Trace/Configuration/ImmutableTracerSettings.cs

```diff
@@ -0,0 +1,333 @@
+#pragma warning disable 618 // App analytics is deprecated, but still used
+            AnalyticsEnabled = settings.AnalyticsEnabled;
+#pragma warning restore 618
+            LogsInjectionEnabled = settings.LogsInjectionEnabled;
+            MaxTracesSubmittedPerSecond = settings.MaxTracesSubmittedPerSecond;
+            CustomSamplingRules = settings.CustomSamplingRules;
+            GlobalSamplingRate = settings.GlobalSamplingRate;
+            Integrations = new ImmutableIntegrationSettingsCollection(settings.Integrations, settings.DisabledIntegrationNames);
+            GlobalTags = new ReadOnlyDictionary<string, string>(settings.GlobalTags);
+            HeaderTags = new ReadOnlyDictionary<string, string>(settings.HeaderTags);
+            DogStatsdPort = settings.DogStatsdPort;
+            TracerMetricsEnabled = settings.TracerMetricsEnabled;
+            RuntimeMetricsEnabled = settings.RuntimeMetricsEnabled;
+            PartialFlushEnabled = settings.PartialFlushEnabled;
+            PartialFlushMinSpans = settings.PartialFlushMinSpans;
+            KafkaCreateConsumerScopeEnabled = settings.KafkaCreateConsumerScopeEnabled;
+            StartupDiagnosticLogEnabled = settings.StartupDiagnosticLogEnabled;
+            HttpClientExcludedUrlSubstrings = settings.HttpClientExcludedUrlSubstrings;
+            HttpServerErrorStatusCodes = settings.HttpServerErrorStatusCodes;
+            HttpClientErrorStatusCodes = settings.HttpClientErrorStatusCodes;
+            ServiceNameMappings = settings.ServiceNameMappings;
+            TraceBufferSize = settings.TraceBufferSize;
+            TraceBatchInterval = settings.TraceBatchInterval;
+            RouteTemplateResourceNamesEnabled = settings.RouteTemplateResourceNamesEnabled;
+        }
+
+        /// <summary>
+        /// Gets the default environment name applied to all spans.
+        /// </summary>
+        /// <seealso cref="ConfigurationKeys.Environment"/>
+        public string Environment { get; }
+
+        /// <summary>
+        /// Gets the service name applied to top-level spans and used to build derived service names.
+        /// </summary>
+        /// <seealso cref="ConfigurationKeys.ServiceName"/>
+        public string ServiceName { get; }
+
+        /// <summary>
+        /// Gets the version tag applied to all spans.
+        /// </summary>
+        /// <seealso cref="ConfigurationKeys.ServiceVersion"/>
+        public string ServiceVersion { get; }
+
+        /// <summary>
+        /// Gets a value indicating whether tracing is enabled.
+        /// Default is <c>true</c>.
+        /// </summary>
+        /// <seealso cref="ConfigurationKeys.TraceEnabled"/>
+        public bool TraceEnabled { get; }
+
+        /// <summary>
+        /// Gets the AdoNet types to exclude from automatic instrumentation.
+        /// </summary>
+        /// <seealso cref="ConfigurationKeys.AdoNetExcludedTypes"/>
+        public HashSet<string> AdoNetExcludedTypes { get; }
```

Comment:
@lucaspimentel I'm assuming you're removing this in https://github.com/DataDog/dd-trace-dotnet/pull/2008, so I'll remove it here too. If not, we should make it immutable instead.

Reply:
I am removing it, yes.

Reply:
Removed in https://github.com/DataDog/dd-trace-dotnet/pull/1951/commits/8c6602d1c19c55084c66a5153fe437ae1fd22257